### PR TITLE
Added integration with Double Slabs

### PR DESCRIPTION
**Pull request by CJMinecraft01**
Added the integration with Double Slabs which allows Crafting Station slabs to be placed on top of each other or be combined with other slabs as well as allowing the slabs to be placed vertically.

The code simply calls a function from my mod which will provide the appropriate TileEntity for the container if the Crafting Station is a vertical slab or a custom double slab. I've also slightly tweaked the method to check if the player can interact with the crafting station to work accordingly.

Please tweak the code as you wish. Should be compatible with DoubleSlabs-1.15-2.6.0

### CODE REVIEW

File: src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java [hunk 1]
```diff
@@ -19,22 +20,65 @@
...
+
+    private static final Method doubleSlabsGetTileEntity;
+    private static final Method doubleSlabsGetBlockState;
+
+    static {
+        Method doubleSlabsGetTileEntity1 = null;
+        Method doubleSlabsGetBlockState1 = null;
+        if (ModList.get().isLoaded("doubleslabs")) {
+            try {
+                Class<?> doubleSlabsFlags = Class.forName("cjminecraft.doubleslabs.Flags");
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java:43]: is it safe to not use a class from an `api` package.

File: src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java [hunk 2]
```diff
@@ -19,22 +20,65 @@
...
+                doubleSlabsGetTileEntity1 = doubleSlabsFlags.getDeclaredMethod("getTileEntityAtPos", BlockPos.class, IBlockReader.class);
+                doubleSlabsGetBlockState1 = doubleSlabsFlags.getDeclaredMethod("getBlockStateAtPos", BlockPos.class, IBlockReader.class);
+            } catch (ClassNotFoundException | NoSuchMethodException e) {
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java:46]: would change this to Throwable and print the stack trace.

File: src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java [hunk 3]
```diff
@@ -19,22 +20,65 @@
...
+import net.minecraft.world.IBlockReader;
 import net.minecraft.world.World;
+import net.minecraftforge.fml.ModList;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 
 public class CraftingStationContainer extends Container {
+
+    private static final Method doubleSlabsGetTileEntity;
```
Comment (shedaniel) [inline src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java:35]: follow the google code guidelines and make these suffix with `_METHOD`

https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

File: src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java (hunk 1 above)
Comment (CJMinecraft01) [reply src/main/java/me/shedaniel/istations/containers/CraftingStationContainer.java:43]: Yep that is completely fine. Perhaps I should have moved it into my API package. It was intended to be used by other mods. Sorry about that

**Comment by shedaniel** [comment]
Sorry for the delay, I have forgot about this completely.

**Pull request merged**